### PR TITLE
Relax some eslint rules for Fiori TypeScript projects

### PR DESCRIPTION
**Pull request by devinea**
Relax some  overly strict eslint rules for Fiori TypeScript type checking for projects
These eslint type checking was recently introduced in https://github.com/SAP/open-ux-tools/pull/1820/files#diff-9df0c3a37ae50def5595af47d18611f9e0a023feffda9ac694889035c990d6f2R22
However it was a breaking change for existing TypeScript projects. This changes from errors to warnings some of the rules in `@typescript-eslint/recommended-requiring-type-checking`.
Full list of rules are here:
https://github.com/typescript-eslint/typescript-eslint/blob/51102482255824f335c38e8c02d88f00e2628c13/packages/eslint-plugin/src/configs/recommended-type-checked.ts#L12-L54

### CODE REVIEW

**Review by IainSAP — DISMISSED** [review]
Thanks @devinea 

- Small lint change to change errors to warn
- No other updates expected

**Comment by longieirl** [comment]
@devinea can you explain why they are required? Are these errors by default?

File: packages/preview-middleware-client/.eslintrc.js [hunk 1]
```diff
@@ -21,10 +21,7 @@ module.exports = {
                 varsIgnorePattern: '^_',
                 argsIgnorePattern: '^_'
             }
-        ],
-        '@typescript-eslint/no-unsafe-argument': 'warn',
-        '@typescript-eslint/no-unsafe-member-access': 'warn',
-        '@typescript-eslint/no-unsafe-assignment': 'warn'
```
Comment (Klaus-Keller) [inline packages/preview-middleware-client/.eslintrc.js:27]: @devinea I think we should keep the rules as they are and work on fxing them. We could consider relaxing the rules for test cases, like done on root: https://github.com/SAP/open-ux-tools/blob/b5277ea58890472c7aab0950a6deb8e101db1730/.eslintrc#L59-L108

Comment (devinea) [reply packages/preview-middleware-client/.eslintrc.js]: Reverted in https://github.com/SAP/open-ux-tools/pull/1909/commits/46d7ef589df4cf1aae04ec91bd9cb58e4fb37c5e

**Review by Klaus-Keller — APPROVED** [review]
Update of rules looks good, thanks @devinea.

**Review by zdravko-georgiev — APPROVED** [review]

**Review by IainSAP — APPROVED** [review]
Thx @devinea 

- As expected

**Pull request merged**